src/lib.rs

```diff
@@ -318,7 +451,7 @@ impl<T> Bus<T> {
         // we run a separate thread responsible for unparking
         // so we don't have to wait for unpark() to return in broadcast_inner
         // sending on a channel without contention is cheap, unparking is not
-        let (unpark_tx, unpark_rx) = mpsc::channel::<thread::Thread>();
+        let (unpark_tx, unpark_rx) = mpsc::channel::<P>();
```

Comment:
You might find it easier to get rid of the separate wakeup thread while you're at it. It was probably a poor choice in the first place anyway!

Reply:
there is a comment about it.. is the comment untrue? could bench it either way.

Reply:
There was some discussion on IRC (I think it was talchas) who pointed out that the channel send/receive ends up being about similarly expensive. And I think in the `futures` case wakeup is even cheaper.